node/pod.go

```diff
@@ -189,6 +189,13 @@ func (pc *PodController) updatePodStatus(ctx context.Context, podFromKubernetes
 		"old reason": podFromKubernetes.Status.Reason,
 	}).Debug("Updated pod status in kubernetes")
 
+	if !running(&podFromProvider.Status) && (podFromProvider.Status.Phase == corev1.PodFailed || podFromProvider.Status.Phase == corev1.PodSucceeded) {
+		log.G(ctx).Debug("Deleting terminal pod from API server")
+		if err := pc.client.Pods(podFromKubernetes.Namespace).Delete(podFromProvider.Name, nil); err != nil {
```

Comment:
Should this be added to the deletion queue?

Reply:
No. It'll set deletion timestamp, with the default grace period. When the pod is updated on api server, we'll get the api update back from apiserver, and make a callback to deletepod, and add it to the deletion queue.